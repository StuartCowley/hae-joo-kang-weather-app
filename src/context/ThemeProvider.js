/* eslint-disable react/jsx-no-constructed-context-values */
import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const ThemeContext = createContext();

export const useTheme = () => useContext(ThemeContext);

export const ThemeProvider = ({ children }) => {
  const [darkTheme, setDarkTheme] = useState(false);
  return (
    <ThemeContext.Provider value={{ darkTheme, setDarkTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};
ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
