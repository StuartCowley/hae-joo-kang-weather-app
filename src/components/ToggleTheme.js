/* eslint-disable jsx-a11y/label-has-associated-control */
import React from "react";
import { useTheme } from "../context/ThemeProvider";

import "../styles/ToggleTheme.css";

const ToggleTheme = () => {
  const { setDarkTheme } = useTheme();

  const toggleHandler = () => {
    setDarkTheme((prev) => !prev);
  };

  return (
    <>
      <input type="checkbox" id="switch" onClick={toggleHandler} />
      <label htmlFor="switch">ToggleTheme</label>
    </>
  );
};

export default ToggleTheme;
