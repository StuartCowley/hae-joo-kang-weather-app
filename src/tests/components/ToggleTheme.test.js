import React from "react";
import { render } from "@testing-library/react";
import ToggleTheme from "../../components/ToggleTheme";
import { ThemeProvider } from "../../context/ThemeProvider";

describe("ToggleTheme", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ThemeProvider>
        <ToggleTheme />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
