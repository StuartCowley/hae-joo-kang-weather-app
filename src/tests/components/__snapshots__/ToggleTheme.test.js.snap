// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleTheme renders correctly 1`] = `
<DocumentFragment>
  <div
    class="toggle-container"
  >
    <input
      id="switch"
      type="checkbox"
    />
    <label
      for="switch"
    >
      ToggleTheme
    </label>
  </div>
</DocumentFragment>
`;
